.github/workflows/container-build-test-publish.yml

```
@@ -0,0 +1,425 @@
+      docker: ${{ steps.filter.outputs.docker }}
+      # Dynamic matrix outputs
+      containers-matrix: ${{ steps.set-matrix.outputs.containers }}
+      test-shared: ${{ steps.set-matrix.outputs.test-shared }}
+      any-container-changed: ${{ steps.set-matrix.outputs.any-changed }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Needed for file change detection
+
+      - name: Check for file changes
+        uses: dorny/paths-filter@v2
```

actionlint (1.7.7)

the runner of "dorny/paths-filter@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)